mkcodes.py

```diff
@@ -79,43 +113,55 @@ def get_files(inputs):
         elif path.suffix in markdown_extensions:
             yield path, path.parent
 
-def add_inits_to_dir(path):
+
+def add_inits_along_path(from_path, to_path):
     """Recursively add __init__.py files to a directory
-    This compensates for https://bugs.python.org/issue23882 and https://bugs.python.org/issue35617
+    This compensates for https://bugs.python.org/issue23882
+    and https://bugs.python.org/issue35617
     """
-    for child in path.rglob('*'):
-        if child.is_dir():
-            (child / '__init__.py').touch()
-
+    to_path = to_path.expanduser().resolve()
+    from_path = from_path.expanduser().resolve()
+    (to_path / '__init__.py').touch()
+    if to_path == from_path:
+        return
+    if to_path.relative_to(from_path):
+        if from_path != to_path.parent:
+            add_inits_along_path(from_path, to_path.parent)
```

Comment:
Could you explain why you changed this `add_inits_along_path` implementation?

Reply:
originally we added an __init__.py file to every dir from the output base directory to every leaf directory.
Now, we may have some directories that will not have any python files in them - we should not add an __init__.py to those directories as they are not python packages. 

Instead, when outputting a .py file we can reasonably just output an __init__.py from the base output directory to the directory containing a python file.

Since we are just using touch to create blank files, there's no real point in checking to see if they are already there.